coalib/results/result_actions/GeneratePatchesAction.py

```diff
@@ -0,0 +1,89 @@
+from coalib.results.result_actions.ShowPatchAction import ShowPatchAction
+from coalib.output.printers.LogPrinter import LogPrinter
+from coalib.parsing.FilterHelper import FilterHelper
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+from coalib.coala_modes import mode_normal
+from coalib.parsing.DefaultArgParser import default_arg_parser
+
+
+def show_possibilities(console_printer, i, action):
+    console_printer.print('[{:>4}]  {}. Apply patch (\'{}\')'.format('', i,
+                                                                     action))
+
+
+class GBear():
+    def __init__(self, bear: str):
+        self.bear = bear
+        self.name = bear
+
+
+class GeneratePatchesAction(ResultAction):
+    SUCCESS_MESSAGE = 'Patch generated successfully.'
+
+    is_applicable = staticmethod(ShowPatchAction.is_applicable)
+
+    def apply(self,
+              result,
+              original_file_dict,
+              file_diff_dict):
+        """
+        (G)enerate patches
+        """
+
+        section = Section('cli')
+        section.add_or_create_setting(Setting('default_actions',
+                                              '*: ApplyPatchAction'))
+        section.add_or_create_setting(Setting('no_config', True))
+
+        console_printer = ConsolePrinter()
+        possible_options = []
+        log_printer = LogPrinter()
+        bears_app = []
+        args = default_arg_parser().parse_args()
+        args.files = []
+        to_filename = sorted(result.diffs.items())[0][0]
+        language = guess_lexer_for_filename(to_filename, 'Error, no file '
+                                                         'found').name
+
+        section.add_or_create_setting(Setting('files', to_filename))
+        path_dir = Path(to_filename)
+        path_dir = path_dir.parts[len(path_dir.parts) - 2]
+        path = to_filename
+        path = path.rstrip(os.sep)
+
+        filtered_bears = FilterHelper.apply_filters(
+            [['language', language]], None)[0]['cli']
+
+        filtered_bears = list(filtered_bears)
+        filtered_bears.insert(0, GBear('SpaceConsistencyBear'))
+        for b in filtered_bears:
+            possible_options.append(b.name)
+
+        console_printer.print('[{:>4}] *0. Do Nothing'.format(''))
+        for i, action in enumerate(possible_options, 1):
+            show_possibilities(console_printer, i, action)
+
+        choose_action = str(input('[{:>4}]  Enter a number: '.format('')))
+        if choose_action is '' or choose_action is '0':
+            return False
+
+        choose_action = int(choose_action)
+        chosed_bear = possible_options[choose_action - 1]
+
+        args.default_actions = '*: ApplyPatchAction'
+
+        bears_app.append(chosed_bear)
+        args.files.append(os.path.basename(path))
+        args.bears = bears_app
+
+        mode_normal(console_printer, log_printer, args, debug=False)
```

Comment:
right now its not readable, it has no comments, probably only you understand the flow